### ISSUE TITLE
SwiftDoc only generates documentation for some non-public members

### ISSUE THREAD

**Issue by a2** [SwiftDocOrg/swift-doc#136]
It seems that SwiftDoc only generates documentation for `public` types but I have noticed that some non-public members are sometimes still generated:

1. `internal` nested type has an entry on the `public` parent type's page but the destination is a 404
2. `internal` properties on `public` types
3. `internal` initializers on `public` types

```swift
public struct MyStruct: Encodable {
    enum CodingKeys: String, CodingKey { // [1]
        case value = "Value"
    } 

    let value: String // [2]

    init(value: String) { // [3]
        self.value = value
    }
}
```

**Comment by mattt** [comment]
Thanks so much for digging in and writing this up, @a2. It's as actionable a bug report as I've ever seen!

If you looked through the source and saw the gnarly bit of code responsible (or have ever tried to wrap your head around Swift ACL rules yourself), the existence of bugs like this should come as little surprise. I'll be sure to add tests around these cases and fix this for the forthcoming release.

**Comment by mattt** [comment]
Fixed by #131.
